packages/components/src/lib/inputUtils.ts

```diff
@@ -0,0 +1,21 @@
+import { SqInput } from "@quri/squiggle-lang";
+
+// Some input defaults are not strings, but we need to pass strings to Squiggle.
+export function defaultAsString(row: SqInput): string {
+  if (typeof row.default === "string") {
+    return row.default;
+  } else if (typeof row.default === "boolean") {
+    return row.default.toString();
+  } else {
+    throw new Error("Invalid default value.");
+  }
+}
+
+// Inputs give us strings, but we need to wrap them in quotes so that Squiggle knows they are meant as strings instead of variables.
+export function alterCodeForSquiggleRun(input: SqInput, code: string) {
```

Comment:
This special case adds significant complexity, and looks like evidence that current Input abstraction is leaking.

Ideally, all input components (form fields) should return the same value type, so that we won't need any extra code for some of them.

But any specific suggestion I could have here is quite a bit of work, and I'm not sure if the return type should be code strings or SqValues (i.e. fields could evaluate their inputs themselves). Not a problem if we do that later.

Reply:
Yea, I agree. I found the two items in this file fairly annoying and non-elegant, but couldn't think of anything else, without a lot more code+complexity added.